common/models/role.js

```diff
@@ -289,7 +290,11 @@ module.exports = function(Role) {
     var resolver = Role.resolvers[role];
     if (resolver) {
       debug('Custom resolver found for role %s', role);
-      resolver(role, context, callback);
+
+      var promise = resolver(role, context, callback);
+      if (promise && typeof promise.then === 'function') {
+        promise.then(callback, callback);
```

Comment:
This will report `resolve` result the same way as `reject` result.

The correct version:

```js
promise.then(
  function(result) { callback(null, result),
  callback // this is the (almost) the same as function(err) { callback(err); }
);
```